DependencyInjection/MonologExtension.php

```diff
@@ -419,30 +419,56 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
 
             if (isset($handler['activation_strategy'])) {
                 $activation = new Reference($handler['activation_strategy']);
-            } elseif (!empty($handler['excluded_404s'])) {
-                if (class_exists(HttpCodeActivationStrategy::class)) {
-                    @trigger_error('The "excluded_404s" option is deprecated in MonologBundle since version 3.4.0, you should rely on the "excluded_http_codes" option instead.', E_USER_DEPRECATED);
-                }
-                $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
-                    new Reference('request_stack'),
-                    $handler['excluded_404s'],
+            }
+            else {
+                $activationDef = new Definition('Monolog\Handler\FingersCrossed\ErrorLevelActivationStrategy', [
                     $handler['action_level']
                 ]);
-                $container->setDefinition($handlerId.'.not_found_strategy', $activationDef);
-                $activation = new Reference($handlerId.'.not_found_strategy');
+                $container->setDefinition($handlerId.'.error_level_strategy', $activationDef);
+                $activation = new Reference($handlerId.'.error_level_strategy');
+            }
+
+            $isLegacyBridge = !\class_exists('Symfony\Bridge\Monolog\Processor\SwitchUserTokenProcessor');
+            if (!empty($handler['excluded_404s'])) {
```

Comment:
All this code is not in the `else` part of the `if (isset($handler['activation_strategy'])) {` anymore, which looks broken to me, as it changes the way the config is processed now when multiple things are passed.

Reply:
well, looks like you handle it by re-adding more `isset($handler['activation_strategy'])` checks below, but this makes the code harder to understand.

Reply:
Thanks for the review, I restored the old blocks